WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -1101,8 +1104,30 @@ private extension ProductFormViewController {
         guard let site = ServiceLocator.stores.sessionManager.defaultSite else {
             return
         }
-        let viewModel = BlazeWebViewModel(source: .productMoreMenu, siteURL: site.url, productID: product.productID)
-        let webViewController = AuthenticatedWebViewController(viewModel: viewModel)
+        viewModel.checkIfBlazeIntroViewIsNeeded()
+
+        if viewModel.shouldShowBlazeIntroView {
+            let blazeCampaignIntroView = BlazeCampaignIntroView(onStartCampaign: {
+                self.dismiss(animated: true)
+                self.viewModel.updateShouldShowBlazeIntroView(to: false)
+                self.navigateToBlazeCampaignCreation(siteUrl: site.url)
+            }, onDismiss: {
+                self.dismiss(animated: true)
+                self.viewModel.updateShouldShowBlazeIntroView(to: false)
+            })
+
+            let hostingController = UIHostingController(rootView: blazeCampaignIntroView)
+            hostingController.modalPresentationStyle = .fullScreen
```

Comment:
Oh, nice catch. I went back to Figma and it seems to use `.pageSheet`, which I added on https://github.com/woocommerce/woocommerce-ios/pull/11011/commits/5bb4f84d127120cbc44bba764aad0e27e440d036

I am still unsure here because `.pageSheet` for some reason has not been used in the codebase before this, so let me know if this is correct.

Reply:
`.automattic` is the default value for `modalPresentationStyle`, we usually don't explicitly set the style and let the system decide what's best.

Reply:
Perfect! good to know https://github.com/woocommerce/woocommerce-ios/pull/11011/commits/ba5e7b51d06768f133077343d6d8361e309b0fd0